### PR TITLE
feat: Remove launch configuration support, add support for ignoring `desired_capacity`

### PR DESCRIPTION
**Pull request by bryantbiggs**
## Description
- Remove support for launch configuration
- Add additional autoscaling group `idc` (ignore desired changes) as well as a new variable `ignore_desired_capacity_changes` to allow  users to opt into an autoscaling group that will ignore `desired_capacity`
- Update variables to expand use of `lt_xxx` abbreviation to `launch_template_xxx`
- See [UPGRADE-5.0.md](https://github.com/bryantbiggs/terraform-aws-autoscaling/blob/breaking-change/remove-launch-config/UPGRADE-5.0.md) for comprehensive list of changes

## Motivation and Context
- Users have often asked for the addition of ignoring `desired_capacity` but unfortunately it cannot be parameterized. With the removal of launch configurations, it seems reasonable enough to add in this additional bit of logic in order to accommodate both options
- There has also been confusion over which settings are launch config vs launch template (user data is the most common one); this change removes any of that ambiguity and confusion
- In addition to the statement above, with these changes we are now able to replace a lot of the duplicated logic in https://github.com/terraform-aws-modules/terraform-aws-eks/pull/1680 . If this change is merged, I intend to do the following:
	- Update the [`self_managed_node_group` sub-module](https://github.com/bryantbiggs/terraform-aws-eks/blob/refactor/breaking-changes/modules/self-managed-node-group/main.tf) to remove the launch template and autoscaling group configurations and replace with this module. This is now possible due to the addition of `ignore_changes  = [desired_capacity]`
	- Update the [`eks_managed_node_group` sub-module](https://github.com/bryantbiggs/terraform-aws-eks/blob/refactor/breaking-changes/modules/eks-managed-node-group/main.tf) to remove the launch template configuration and replace with this module. In hindsight, there is no need to duplicate the custom launch template configuration there while this module provides the means to create a standalone launch template without an autoscaling group

With these two changes, we are then able to offload even more complexity and logic from the core EKS module and centralize the changes in the appropriate modules (i.e. - if a new feature is added to launch templates then we can update this module and propagate to the EKS module)

## Breaking Changes
- Yes!  Launch configuration support is no longer provided now that AWS has stated that they are no longer adding/updating the APIs for launch configurations (launch templates are the successor). In its place, a second autoscaling group that is identical to the default has been added - the only difference between the two is that the second ASG now ignores `desired_capacity` changes to allow users to utilize autoscaling without reverting or showing up in Terraform plans/applies.

## How Has This Been Tested?
- [x] I have tested and validated these changes using one or more of the provided `examples/*` projects

### CODE REVIEW

**Comment by bryantbiggs** [comment]
Shhhh

**Comment by antonbabenko** [comment]
@bryantbiggs 
Ops, I see that this one should have been merged around the same time as #179 . Could you finish this one and release version 5.1 right away?

**Comment by bryantbiggs** [comment]
ok should be all set @antonbabenko

File: UPGRADE-5.0.md [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+## Additional changes
+
+### Added
+
+- A second autoscaling group `idc` (ignore desired capacity) has been added to allow users to utilize an autoscaling group that will ignore changes to `desired_capacity`
+
+### Modified
+
+- `var.tags` was previously a list of maps of `key`, `value`, and `propagate_at_launch`. It has been changed to a standard map of `key` = `value` pairs
+- The use of `for_each = lookup(map.value, "attr", null) != null ? [map.value.attr] : []` pattern has been replaced with a more concise, readable pattern of `for_each = can(map.value.attr) ? [map.value.attr] : []`. This is a no-op change for users
```
Comment (antonbabenko) [inline UPGRADE-5.0.md:22]: Good to know.

Can it be written like this - `for_each = try([map.value.attr], [])` ?

Comment (antonbabenko) [reply UPGRADE-5.0.md:22]: Will it work? I am going to merge this one like this anyway.

**Comment by antonbabenko** [comment]
This PR is included in version 5.1.0 :tada:

**Pull request merged**